lib/matplotlib/axes/_base.py

```diff
@@ -3016,7 +3016,8 @@ def _validate_converted_limits(self, limit, convert):
                 raise ValueError("Axis limits cannot be NaN or Inf")
             return converted_limit
 
-    def set_xlim(self, left=None, right=None, emit=True, auto=False, **kw):
+    def set_xlim(self, left=None, right=None, emit=True, auto=False,
+                 *, xmin=None, xmax=None):
```

Comment:
Would it be worth deprecating xmin and xmax or left or right? Seems a bit silly having two ways of specifying what is really the same argument.

Reply:
+1 that two ways are a bit silly. Not sure if we have to go as far as deprecating (but wouldn‘t object either). Anyway, there should be one recommended way and the other only is only a backward compatible fallback.

The question is which to recommend:
- xmin, xmax is more generic and easy to transfer, e.g. ymin rmin 
- left, right is more semantic

To me +0.5 for keeping xmin, xmax

Reply:
Actually, left/right is clearly better IMO: one can pass left > right for inverted axes, which sounds less silly that xmin > xmax...

Reply:
This code path is due to an API change from @efiring in 2010 (9ca5db03f692e73d155dbc7ca752413ccee01df6) and this has not been _documented_ in the docstring.    I am inclined to deprecate these unless we find that they are widely used in the wild.

I am also :+1: on pushing the deprecation work to a different PR.